src/arrayMethodSort.js

```diff
@@ -3,9 +3,24 @@
 /**
  * Implement method Sort
  */
+function defaultCompare(a, b) {
+  return String(a) > String(b) ? 1 : -1;
+}
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 1; j < this.length - i; j++) {
+        if (compareFunction(this[j - 1], this[j]) > 0) {
+          const item = this[j - 1];
+
+          this[j - 1] = this[j];
+          this[j] = item;
+        }
+      }
+    }
```

Comment:
The current implementation of the sorting algorithm is a bubble sort, which has a time complexity of O(n^2). This is not efficient for larger arrays. While it's fine for an exercise, in production code, you'd typically want to use more efficient algorithms like merge sort or quick sort.